### ISSUE TITLE
bc2pg - specify geometry type

### ISSUE THREAD

**Issue by smnorris** [smnorris/bcdata#92]
I did not see a way to determine if a WFS response will contain multipart or singlepart, and for many datasets it seems WFS will return a mix of singlepart/multipart. This is handled fine by postgis by just loading to a generic 'Geometry' type. 

However, QGIS doesn't like this - it will scan a table to determine the type and this can take a very long time for large tables.

Currently for tables I'm using in QGIS I'll `--promote_to_multi` on load to ensure everything is multipart, then `ALTER TABLE` and set type to something like `geometry(MULTIPOLYGON, 3005)` It would be nice if `bc2pg` could do this automatically.

**Comment by smnorris** [comment]
New function `bcdata.get_type` will find the geometry type of first record in the table.  With this known, we can do this:
```
        if geometry_type.upper() in ["POINT", "MULTIPOINT"]:
            nlt = "MULTIPOINT"
        elif geometry_type.upper() in ["LINESTRING", "MULTILINESTRING"]:
            nlt = "MULTILINESTRING"
        elif geometry_type.upper() in ["POLYGON", "MULTIPOLYGON"]:
            nlt = "MULTIPOLYGON"
```
Which will work most of the time... but will bail on features with Z or M values. I think it is safe to presume no features will have M values but some features do have Z values. Further investigation of the feature requested in `get_type` should find that, something like this:

```
if len(r.json()['features'][0]['geometry']['coordinates'][0]) == 3:
  z_dim is true
```

**Comment by ateucher** [comment]
Side question @smnorris - can you give an example of a record with a values? Would be good to test against.

**Comment by smnorris** [comment]
You mean z values? `WHSE_BASEMAPPING.FWA_STREAM_NETWORKS` are 3d. I think some of the other FWA tables might be too.

```
import bcdata
import requests

parameters = {
  "service": "WFS",
  "version": "2.0.0",
  "request": "GetFeature",
  "typeName": "WHSE_BASEMAPPING.FWA_STREAM_NETWORKS_SP",
  "outputFormat": "json",
  "count": 1,
}
r = requests.get(bcdata.WFS_URL, params=parameters)
r.json()['features'][0]['geometry']

>>>{'type': 'LineString', 'coordinates': [[706013.865, 1559813.368, 1538], [706012.32, 1559802.511, 1540.72], [706009.965, 1559791.555, 1544.808], [706008.347, 1559778.637, 1549.558], [706007.968, 1559769.604, 1552.856], [706007.713, 1559761.674, 1555.751], [706007.206, 1559748.676, 1560.608], [706007.006, 1559742.65, 1562.954], [706006.592, 1559734.732, 1566.076], [706007.344, 1559728.638, 1568.319], [706008.993, 1559720.571, 1571.331], [706012.682, 1559709.336, 1576.032], [706016.231, 1559698.271, 1580.698], [706016.983, 1559692.176, 1583.039], [706018.631, 1559684.109, 1585.981], [706019.276, 1559677.068, 1588.348], [706020.925, 1559669.001, 1590.895], [706021.694, 1559663.064, 1592.643], [706023.377, 1559653.882, 1595.046], [706024.863, 1559642.966, 1597.882], [706024.697, 1559635.825, 1599.72], [706024.263, 1559624.888, 1602.537], [706025.731, 1559613.814, 1604]]}
```

**Comment by ateucher** [comment]
Sorry, typo - I did mean z. Thanks!

**Comment by smnorris** [comment]
the added `--nlt` option isn't reliable, remove and revert to previous workaround of just letting ogr write to a general geometry type and/or promoting to multipart

**Comment by smnorris** [comment]
fixed with v0.5.1
